src/engine/blocks.js

```diff
@@ -914,6 +914,33 @@ class Blocks {
         }
     }
 
+    /**
+     * Update blocks after a variable gets renamed.
+     * Any block referring to the old name of the asset should get updated
+     * to refer to the new name.
+     * @param {string} oldName The old name of the asset that was renamed.
+     * @param {string} newName The new name of the asset that was renamed.
+     * @param {string} targetName The name of the target the variable belongs to.
+     */
+    updateVariableName (oldName, newName, targetName) {
+        const blocks = this._blocks;
+        for (const blockId in blocks) {
+            const block = this.getBlock(blockId);
+            if (block &&
+                block.fields.hasOwnProperty('PROPERTY') &&
```

Comment:
```suggestion
            if (!Object.prototype.hasOwnProperty.call(blocks, blockId)) continue;
            const block = blocks[blockId];
            if (block.fields.hasOwnProperty('PROPERTY') &&
```

I think this is a bit easier to follow:
- `getBlock` is just a lookup on `this._blocks` which we're already iterating over.
- Instead of checking that `block` exists, check if `this._blocks` has the block ID.

Unless there's a scenario in which `block` exists and is null?